### ISSUE TITLE
"find_dependency" error in CeresConfig.cmake

### ISSUE THREAD

**Issue by LarryDong** [ceres-solver/ceres-solver#533]
Hello everyone,
When I use CMakeLists.txt to complie my program, which used " find_package(Ceres REQUIRED)", 
An error occurred: 
```
CMake Error at /usr/share/cmake-3.5/Modules/CMakeFindDependencyMacro.cmake:45 (message):
  Invalid arguments to find_dependency
Call Stack (most recent call first):
  /usr/local/lib/cmake/Ceres/CeresConfig.cmake:175 (find_dependency)
  CMakeLists.txt:14 (find_package)
```

I've checked the **CeresConfig.cmake** file, and find that **${CERES_EIGEN_VERSION}** is 3.3.7, which is exactly my eigen3 version.

I need your help, I'm not sure whether that it's my CMakeLists.txt error or I should modify CeresConfig.cmake, or my eigen3's problem? 
Thanks again.

**Comment by alexsmac** [comment]
Where is Eigen installed?  Some recent updates to Ceres' detection of
dependencies have been made to make Ceres more relocatable, which means
that the paths are no longer hard-coded into CeresConfig.cmake, but instead
we rely on the packaged config files for dependencies such as Eigen that
supply them.  However, this means that you will have to specify Eigen3_DIR
when building your own project that uses Ceres if it is installed in a
non-standard location that is not searched by CMake by default -
specifically, if you had to specify Eigen3_DIR when building Ceres as it
was otherwise not found, you will now also have to specify it when
compiling your own project that uses Ceres.

**Comment by LarryDong** [comment]
@alexsmac 
It just hit me that the error, " Invalid arguments to find_dependency", So I checked cmake, and found out that the parameters of "find_dependency" has been changed after cmake3.5. 
Now I updated cmake to 3.16 and everything goes on well.
Still thanks for your quick reply.

Maybe this issue can be a reminder for others because I didn't realize that my cmake is out of date, but still the newest from apt.

**Comment by LarryDong** [comment]
Issue closed. Pay attention to CMAKE version if others have  the same problem.

**Comment by tipf** [comment]
I think it would be better to reopen this Issue, because the main CMakeLists.txt currently states:
`cmake_minimum_required(VERSION 3.5)`
This could be raised of course, but this may cause some problems on older systems. (e.g. Ubuntu 16.04)

**Comment by alexsmac** [comment]
This should now be fixed in master, @LarryDong please can you confirm

**Comment by LarryDong** [comment]
Thanks, @alexsmac. Now the -master goes on well but I use CMAKE-3.16 now and don't want to switch back to 3.5. I think there should be no more problems after changing to find_package(). Thanks for your quick reply.

**Comment by tipf** [comment]
I can confirm that the issue is fixed for CMake 3.5. Thanks @alexsmac !

**Comment by salihkaragoz** [comment]
+1
